File: Carros Objective-C/Carros Objective-C/Model/Utils/VideoUtil.m
//
//  VideoUtil.m
//  Carros Objective-C
//
//  Created by Antonio Lopes on 12/15/16.
//  Copyright © 2016 Antonio Lopes. All rights reserved.
//

#import "VideoUtil.h"

@implementation VideoUtil

// Reproduz o vídeo com um player utilizando uma view como conteiner
- (void) playFile:(NSString *)file withView:(UIView *)view {
    NSString * path = [[NSBundle mainBundle] pathForResource:file ofType:nil];
    if (path != nil) {
        NSURL * url = [NSURL fileURLWithPath:path];
        [self playUrl:url withView:view];
    }
}

// Reproduz o vídeo com um player utilizando uma view como conteiner
- (void) playUrl:(NSURL *)url withView:(UIView *)view {
    // cria o componente que gerencia o vídeo
    if (player == nil) {
        if (playerVC == nil) {
            playerItem = [[AVPlayerItem alloc] initWithURL:url];
            player =  [[AVPlayer alloc] initWithPlayerItem:playerItem];
            
            playerVC = [[AVPlayerViewController alloc] init];
            playerVC.player = player;
            playerVC.showsPlaybackControls = TRUE;
            playerVC.view.frame = view.frame;
            [[playerVC view] sizeToFit];
            playerVC.videoGravity = AVLayerVideoGravityResizeAspect;
            
            [view addSubview:playerVC.view];
            
            [playerVC.player play];
        }
    }
    // inicia o video
    [player play];
}

- (void) pause {
    [player pause];
}

- (void) stop {
    [player pause];
}

// Reproduz o filme em tela cheia
- (void) playFileFullScreen:(NSString *)file withViewController:(UIViewController *)viewController {
    // converte o nome do arquivo em URL
    NSString * path = [[NSBundle mainBundle] pathForResource:file ofType:nil];
    if (path != nil) {
        NSURL * url = [NSURL fileURLWithPath:path];
        [self playUrlFullScreen:url withViewController:viewController];
    }
}

- (void) playUrlFullScreen:(NSURL *)url withViewController:(UIViewController *)viewController {
    // cria o controller especializado em reproduzir vídeo
    if (playerVC == nil) {
        playerItem = [[AVPlayerItem alloc] initWithURL:url];
        player = [[AVPlayer alloc] initWithPlayerItem:playerItem];
        
        playerVC = [[AVPlayerViewController alloc] init];
        playerVC.player = player;
        playerVC.showsPlaybackControls = TRUE;
        playerVC.videoGravity = AVLayerVideoGravityResizeAspect;
        
        [viewController presentViewController:playerVC animated:TRUE completion: ^(void) {
            [playerVC.player play];
        }];
    }
}

@end

File: Carros Swift/Carros Swift/Model/Utils/VideoUtil.swift
//
//  VideoUtil.swift
//  Carros Swift
//
//  Created by Antonio Lopes on 12/15/16.
//  Copyright © 2016 Antonio Lopes. All rights reserved.
//

import Foundation
import MediaPlayer
import AVKit

class VideoUtil {
    
    var playerVC : AVPlayerViewController!
    var playerItem : AVPlayerItem!
    var player : AVPlayer!
    var playerLayer : AVPlayerLayer!
    
    // Reproduz o vídeo com um player utilizando uma view como conteiner
    func playFile(file: String, view: UIView) {
        
        if let path = Bundle.main.path(forResource: file, ofType: nil)  {
            let url = NSURL.fileURL(withPath: path)
            self.playUrl(url: url, view: view)
        }
    }
    
    // Reproduz o vídeo com um player utilizando uma view como conteiner
    func playUrl(url: URL, view: UIView) {
        // cria o componente que gerencia o vídeo
        if (self.player == nil) {
            if (self.playerVC == nil) {
                playerItem = AVPlayerItem.init(url: url)
                player = AVPlayer.init(playerItem: playerItem)
                
                playerVC = AVPlayerViewController.init()
                playerVC.player = player
                playerVC.showsPlaybackControls = true
                playerVC.view.frame = view.frame
                playerVC.view.sizeToFit()
                playerVC.videoGravity = AVLayerVideoGravityResizeAspect
                
                view.addSubview(playerVC.view)
                
                self.playerVC.player?.play()
            }
        }
        // inicia o video
        self.player!.play()
    }
    
    func pause() {
        self.player!.pause()
    }
    
    func stop() {
        self.player!.pause()
    }
    
    // Reproduz o filme em tela cheia
    func playFileFullScreen(file: String, viewController: UIViewController) {
        // converte o nome do arquivo em URL
        if let path = Bundle.main.path(forResource: file, ofType: nil)  {
            let url = NSURL.fileURL(withPath: path)
            self.playUrlFullScreen(url: url, viewController: viewController)
        }
    }
    
    func playUrlFullScreen(url: URL, viewController: UIViewController) {
        // cria o controller especializado em reproduzir vídeo
        if (self.playerVC == nil) {
            playerItem = AVPlayerItem.init(url: url)
            player = AVPlayer.init(playerItem: playerItem)
            
            playerVC = AVPlayerViewController.init()
            playerVC.player = player
            playerVC.showsPlaybackControls = true
            playerVC.videoGravity = AVLayerVideoGravityResizeAspect
            
            viewController.present(playerVC, animated: true, completion: { () -> Void in
                self.playerVC.player?.play()
            })
            
        }
    }
}
